### PR TITLE
Add more classes

### PR DESCRIPTION
**Pull request by Daphineatim**
### In this milestone, the following were implemented;

- [ ] Created a class Nameable.

- Implemented a method called correct name that will raise a NotImplementedError.

- [ ] Made sure that your Person class inherits from Nameable.

- Made sure that this class has a method correct_name implemented. It should simply return the name attribute.

- [ ] Made sure that the base decorator inherits from Nameable.

- assigned a nameable object from params to an instance variable in the constructor.

- Implemented the correct_name method that returns the result of the correct_name method of the @nameable.

- [ ] Prepared CapitalizeDecorator and TrimmerDecorator

For the CapitalizeDecorator:
- Created a class that inherits from the base Decorator class.

- Implement a method correct_name that capitalizes the output of @nameable.correct_name.

For the TrimmerDecorator:
- Created a class that inherits from the base Decorator class.

- Implemented a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

Thank you for viewing my project!

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,8 +1,11 @@
...
+
+class Person < Nameable
   attr_accessor :name, :age
   attr_reader :id
 
   def initialize(age, name = 'Unknown', parent_permission: true)
+    super()
     @id = Random.rand(1..100)
     @age = age
     @name = name
```
Comment (mckent05) [inline person.rb:11]: - According to the project requirements, this class should have a `correct_name ` method that returns the name attribute 🙏

File: trimmer_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require './nameable'
+
+class TrimmerDecorator < Nameable
+  def initialize(nameable = Nameable.new)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name.slice(0, 10)
```
Comment (mckent05) [inline trimmer_decorator.rb]: - This method should return the trimmed name if the name is longer than 10 chars, else it should return the name. Please add a condition to return the name if it is not more than 10 chars 👍 👍

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @Daphineatim, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (Daphineatim) [reply person.rb:11]: Thank you, I have worked on the changes.

File: trimmer_decorator.rb (hunk 2 above)
Comment (Daphineatim) [reply trimmer_decorator.rb]: Thank you, I have worked on it.

**Review by sja-thedude — APPROVED** [review]
## Project Approved 👍 

Hi @Daphineatim, This is @sja-thedude your code reviewer.

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights:
- All linters passing
- All requirements completed
- Professional readme

giphy

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**